qiskit/aqua/operators/gradients/circuit_qfis/overlap_block_diag.py

```diff
@@ -0,0 +1,305 @@
+        dag = circuit_to_dag(circuit)
+        dag_layers = ([i['graph'] for i in dag.serial_layers()])
+        num_qubits = circuit.num_qubits
+        layers = list(
+            zip(dag_layers, [{x: False for x in range(0, num_qubits)} for layer in dag_layers]))
+
+        # initialize the ledger
+        # The ledger tracks which qubits in each layer are available to have
+        # gates from subsequent layers shifted backward.
+        # The idea being that all parameterized gates should have
+        # no descendants within their layer
+        for i, (layer, ledger) in enumerate(layers):
+            op_node = layer.op_nodes()[0]
+            is_param = op_node.op.is_parameterized()
+            qargs = op_node.qargs
+            indices = [qarg.index for qarg in qargs]
+            if is_param:
+                for index in indices:
+                    ledger[index] = True
+
+        def apply_node_op(node, dag, back=True):
+            op = copy.copy(node.op)
+            qargs = copy.copy(node.qargs)
+            cargs = copy.copy(node.cargs)
+            condition = copy.copy(node.condition)
+            if back:
+                dag.apply_operation_back(op, qargs, cargs, condition)
+            else:
+                dag.apply_operation_front(op, qargs, cargs, condition)
+
+        converged = False
+
+        for _ in range(dag.depth() + 1):
+            if converged:
+                break
+
+            converged = True
+
+            for i, (layer, ledger) in enumerate(layers):
+                if i == len(layers) - 1:
+                    continue
+
+                (next_layer, next_ledger) = layers[i + 1]
+                for next_node in next_layer.op_nodes():
+                    is_param = next_node.op.is_parameterized()
+                    qargs = next_node.qargs
+                    indices = [qarg.index for qarg in qargs]
+
+                    # If the next_node can be moved back a layer without
+                    # without becoming the descendant of a parameterized gate,
+                    # then do it.
+                    if not any([ledger[x] for x in indices]):
+
+                        apply_node_op(next_node, layer)
+                        next_layer.remove_op_node(next_node)
+
+                        if is_param:
+                            for index in indices:
+                                ledger[index] = True
+                                next_ledger[index] = False
+
+                        converged = False
+
+                # clean up empty layers left behind.
+                if len(next_layer.op_nodes()) == 0:
+                    layers.pop(i + 1)
+
+        partitioned_circs = [dag_to_circuit(layer[0]) for layer in layers]
+        return partitioned_circs
+
+    @staticmethod
+    def _sort_params(params):
+        def compare_params(param1, param2):
+            name1 = param1.name
+            name2 = param2.name
+            value1 = name1[name1.find("[") + 1:name1.find("]")]
```

Comment:
It looks like this assumes parameters from a `ParameterVector` (since only they are by default named something like `name[i]`). Can we assure that this is always the case?

Reply:
No, probably not. I will see about making this more robust.

Reply:
@BryceFuller

Reply:
Nvm I deleted this method as it is not used anywhere